### PR TITLE
Game start countdown

### PR DESCRIPTION
**Pull request by Vessalia**

### CODE REVIEW

File: core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        {
+            int ID = e.getComponent(PlayerComponent.class).handle.ID;
+
+            if(ID == currPlayer)
+            {
+                position.set(e.getComponent(PositionComponent.class).position);
+            }
+            if(ID > numPlayers)
+            {
+                numPlayers = ID;
```
Comment (goldDaniel) [inline core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java:47]: This is incorrect and is just coincidence if it works. numPlayers should be equal to entities.size

File: core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import smashdudes.ecs.Entity;
+import smashdudes.ecs.components.*;
+
+import java.lang.reflect.GenericArrayType;
+
+public class CountdownCameraSystem extends GameSystem
+{
+    private final Vector2 position = new Vector2(0, 0);
+    private int numPlayers;
+    private int currPlayer;
```
Comment (goldDaniel) [inline core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java:17]: Maybe store a reference to the playerHandle instead. That way we can't accidentally manipulate it

File: core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        float curr = count.currDuration;
+        float threshold = max / numPlayers;
+        timer += dt;
+        if(timer < max)
+        {
+            if (curr <= max - currPlayer * threshold)
+            {
+                currPlayer++;
+            }
+        }
```
Comment (goldDaniel) [inline core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java:68]: A comment explaining what this is calculating would be appreciated

File: core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        else
+        {
+            entity.removeComponent(CountdownComponent.class);
+            entity.addComponent(new AveragePositionCameraComponent());
+        }
+
+        cam.camera.position.lerp(new Vector3(position, 0), 1/50f);
+        cam.camera.zoom = 1f;//MathUtils.lerp(cam.camera.zoom, MathUtils.clamp(max.dst(min) * 0.1f, 1.4f, 2.6f), 1/50f);
```
Comment (goldDaniel) [inline core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java:76]: Remove the comment

File: core/src/main/java/smashdudes/ecs/systems/CountdownSystem.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            engine.addEvent(new CountdownEvent(entity, 2));
+        }
+        else if(cc.currDuration > 0)
+        {
+            engine.addEvent(new CountdownEvent(entity, 1));
+        }
+        else
+        {
+            engine.addEvent(new CountdownEvent(entity, 0));
+        }
```
Comment (goldDaniel) [inline core/src/main/java/smashdudes/ecs/systems/CountdownSystem.java:51]: These will fire every frame. The events should only fire once each

File: core/src/main/java/smashdudes/ecs/systems/CountdownSystem.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.badlogic.gdx.utils.Array;
+import smashdudes.ecs.Engine;
+import smashdudes.ecs.Entity;
+import smashdudes.ecs.components.CountdownComponent;
+import smashdudes.ecs.events.CountdownEvent;
+
+public class CountdownSystem extends GameSystem
+{
+    private Array<Boolean> eventsHaveFired = new Array<>();
```
Comment (goldDaniel) [inline core/src/main/java/smashdudes/ecs/systems/CountdownSystem.java:11]: Unused

File: core/src/main/java/smashdudes/ecs/systems/UIRenderSystem.java [hunk 7]
```diff
@@ -53,6 +56,9 @@ public int compareTo(Object o)
 
     private final Array<CharacterDisplay> players = new Array<>();
 
+    private String countDisplay = "";
+    private float timer;
```
Comment (goldDaniel) [inline core/src/main/java/smashdudes/ecs/systems/UIRenderSystem.java:60]: What is this a timer for? Rename to gameStartTimer or something like that

File: core/src/main/java/smashdudes/ecs/systems/UIRenderSystem.java [hunk 8]
```diff
@@ -95,6 +105,17 @@ public void updateEntity(Entity entity, float dt)
...
+        if(countDisplay.equals("GO!"))
+        {
+            timer += dt;
+
+            if(timer >= 1)
+            {
+                timer = 0;
+                countDisplay = "";
+            }
+        }
```
Comment (goldDaniel) [inline core/src/main/java/smashdudes/ecs/systems/UIRenderSystem.java:118]: Comment explaining what this is doing

**Review by goldDaniel — CHANGES_REQUESTED** [review]
Some small changes!

File: core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java (hunk 1 above)
Comment (Vessalia) [reply core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java:47]: the statement above it should probably be changed as well then

File: core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java (hunk 2 above)
Comment (Vessalia) [reply core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java:17]: I'm not really sure what you mean

**Review by goldDaniel — APPROVED** [review]

**Pull request merged**